src/client/datascience/interactive-window/nativeInteractiveWindowProvider.ts

```diff
@@ -125,10 +128,30 @@ export class NativeInteractiveWindowProvider implements IInteractiveWindowProvid
         // Ensure all our controllers are registered with VS Code
         await this.notebookControllerManager.loadNotebookControllers();
 
+        // Fetch the active interpreter and use the matching controller
+        const api = await this.pythonApi.getApi();
+        const activeInterpreter = await api.getActiveInterpreter();
+        const preferredController = this.notebookControllerManager
+            .registeredNotebookControllers()
+            .find((controller) => {
+                return (
+                    controller.connection.interpreter?.path === activeInterpreter?.path &&
```

Comment:
Looking up now, but I don't believe this will be correct. Multiple conda envs can share the same interpreter path.

Reply:
Yep you're totally right

Reply:
In startUsingPythonInterpreter type connections I believe that we save the entire env. displayName and envType are optional, but could help to distinguish here.

Reply:
Hmm, I was just looking more at this, I can't actually recall the way that we were seeing conda envs with shared interpreter paths. I recall discussing that but I was unable to make one myself.

Reply:
Ok, I did actually create one. Looking at the code there might actually be a bug here for existing interactive window. Digging in now.

Reply:
I pulled this logic from https://github.com/microsoft/vscode-jupyter/blob/a127a52f71c0bb10a1aa0979a310d871fdeb521d/src/client/datascience/jupyter/kernels/helpers.ts#L388 Does this need to be updated as well?

Reply:
This existing issue might be the same problem: https://github.com/microsoft/vscode-jupyter/issues/5650

Reply:
Yup that's the location that I ran into looking at your code here. I'm building a quick repro to test.